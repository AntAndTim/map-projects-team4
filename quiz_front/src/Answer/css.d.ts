declare module 'csstype' {
    interface Properties {
        '--check-box-color'?: 'green' | 'red';
    }
}