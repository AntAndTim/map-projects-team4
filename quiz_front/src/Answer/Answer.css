.ant-checkbox-checked .ant-checkbox-inner {
    background-color: var(--check-box-color) !important;
    border-color: var(--check-box-color) !important;
}