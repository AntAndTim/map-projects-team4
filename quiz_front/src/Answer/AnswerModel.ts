export class AnswerModel {
    constructor(id: number, text: string, correct: boolean) {
        this.id = id;
        this.text = text;
        this.correct = correct;
    }

    id: number
    text: string
    correct: boolean
}