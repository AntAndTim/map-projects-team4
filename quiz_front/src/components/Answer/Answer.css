.ant-card.Answer {
    /*background-color: var(--check-box-color) !important;*/
    /*border-color: var(--check-box-color) !important;*/

    width: 100%;
    margin-bottom: 16px;
    margin-right: 16px;

    font-size: 20px;
    text-align: center;
}

.ant-card.Answer_selected, .Answer_selected.ant-card-hoverable:hover {
    border: 2px solid #1890ff;
}

.ant-card.Answer_green, .Answer_green.ant-card-hoverable:hover {
    border: 2px solid green;
}

.ant-card.Answer_yellow, .Answer_yellow.ant-card-hoverable:hover {
    border: 2px solid yellow;
}

.ant-card.Answer_red, .Answer_red.ant-card-hoverable:hover {
    border: 2px solid red;
}