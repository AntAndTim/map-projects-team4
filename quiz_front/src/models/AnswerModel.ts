export interface AnswerModel {
    id: number;
    text: string;
    correct: boolean;
    selected?: boolean;
    color?: string;
}