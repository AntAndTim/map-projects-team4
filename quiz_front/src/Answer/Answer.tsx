import {Checkbox} from 'antd';
import React, {useState} from "react";
import {AnswerModel} from "./AnswerModel";
import './Answer.css'

export const Answer = ({answer}: { answer: AnswerModel }) => {
    const [checked, setChecked] = useState(false)
    const [disabled, setDisabled] = useState(false)

    const checkBoxStyle = answer.correct
        ? {'--check-box-color': 'green' as const}
        : {'--check-box-color': 'red' as const};

    return (
        <div>
            <Checkbox
                disabled={disabled}
                checked={checked}
                onChange={(e) => {
                    setChecked(e.target.checked)
                    setDisabled(true)
                }}
                style={checkBoxStyle}
            >
                {answer.text}
            </Checkbox>
        </div>
    );
}