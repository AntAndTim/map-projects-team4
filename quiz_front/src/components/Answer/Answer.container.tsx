import React, {useCallback} from "react";
import {AnswerModel} from "../../models/AnswerModel";
import {Answer} from "./Answer";

interface AnswerProps {
    answer: AnswerModel;
    onAnswerClick: (id: number) => void;
    haveAnswered: boolean;
}

export const AnswerContainer: React.FC<AnswerProps> = ({
  answer,
  onAnswerClick,
  haveAnswered,
}) => {
    const clickHandler = useCallback(() => onAnswerClick(answer.id), [onAnswerClick]);

    return (
        <Answer
            answer={answer}
            onClick={clickHandler}
            haveAnswered={haveAnswered}
        />
    );
}