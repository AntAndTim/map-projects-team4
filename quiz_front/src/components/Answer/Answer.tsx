import {Card} from 'antd';
import React, {useCallback} from "react";
import {AnswerModel} from "../../models/AnswerModel";
import './Answer.css'

interface AnswerProps {
    answer: AnswerModel;
    onAnswerClick: (id: number) => void;
    haveAnswered: boolean;
}

export const Answer: React.FC<AnswerProps> = ({
  answer,
  onAnswerClick,
  haveAnswered,
}) => {
    const onClick = useCallback(() => onAnswerClick(answer.id), [onAnswerClick]);
    const className = ["Answer", answer.selected && "Answer_selected", answer.color && `Answer_${answer.color}`].filter(Boolean).join(' ');

    return (
        <Card
            hoverable={!haveAnswered}
            className={className}
            onClick={onClick}
        >
            {answer.text}
        </Card>
    );
}