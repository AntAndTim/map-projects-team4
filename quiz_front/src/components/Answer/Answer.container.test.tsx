import React from 'react';
import {render} from 'enzyme';
import {AnswerContainer} from "./Answer.container";
import {AnswerModel} from "../../models/AnswerModel";

const answer: AnswerModel = {
    id: 1,
    text: "AnswerText",
    correct: false,
}

describe('AnswerContainer', () => {
    it('Should render component', () => {
        const wrapper = render(
            <AnswerContainer
                answer={answer}
                onAnswerClick={jest.fn}
                haveAnswered={false}
            />
        );

        expect(wrapper).toMatchSnapshot();
    })
});