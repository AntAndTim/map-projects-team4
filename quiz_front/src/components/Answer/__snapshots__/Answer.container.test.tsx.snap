// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerContainer Should render component 1`] = `
<div
  class="ant-card ant-card-bordered ant-card-hoverable Answer"
>
  <div
    class="ant-card-body"
  >
    AnswerText
  </div>
</div>
`;
